### PR TITLE
pkg: Add checkcommits tool.

### LINKED ISSUE #18: Add commit-checking tool
Rewrite https://github.com/01org/cc-oci-runtime/pull/873 in go.

By hosting the tool in this repository, the other repos (virtcontainers, cc-oci-runtime and the clearcontainers ones) can opt to use the tool by having their CI `go get` it.

Supercedes https://github.com/01org/cc-oci-runtime/issues/872.

### PR DESCRIPTION
**Pull request by jodh-intel**
Rework of https://github.com/01org/cc-oci-runtime/pull/873 in golang.

By hosting it in the tests repository our other repos can "go get" if
they wish to use it.

Fixes #18.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: pkg/checkcommits/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+atleast one "Fixes #XXX" entry in the commits and ensuring each commit
+contains a developer sign-off).
+
+## List of Options
+
+For full details, run:
+
+```
+$ ./checkcommits -h
+
```
Comment (dlespiau) [inline pkg/checkcommits/README.md]: extra new line

**Comment by jodh-intel** [comment]
@dlespiau - thanks - fixed.

File: Makefile [hunk 2]
```diff
@@ -18,4 +18,10 @@ check:
 
 all: functional
 
+checkcommits:
```
Comment (dlespiau) [inline Makefile]: looks like this could be added to .gitignore

File: Makefile [hunk 3]
```diff
@@ -18,4 +18,10 @@ check:
...
+	cd pkg/checkcommits/ && go build && go test
```
Comment (dlespiau) [inline Makefile]: Can we add the tests to the usual make check? Sounds like a good idea to execute them at every commit once we've hooked CI to this repo

File: pkg/checkcommits/checkcommits.go [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/urfave/cli"
```
Comment (dlespiau) [inline pkg/checkcommits/checkcommits.go]: Looks like this could use some vendoring.

File: Makefile (hunk 3 above)
Comment (dlespiau) [inline Makefile]: Not sure about putting `checkcommits` in `pkg/`, that's usually reserved for non main packages. Those are put in a `cmd` directory.

**Comment by dlespiau** [comment]
with or without the changes outlined above, I think we can merge this.

lgtm

**Comment by jodh-intel** [comment]
I've addressed @dlespiau's comments. However, `make check` shows I need to refactor `checkCommitBody()` a little as the cyclomatic complexity is `17` (our limit is `15`), hence adding `do-no-merge` label for now.

**Comment by jodh-intel** [comment]
Branch updated.

**Comment by chavafg** [comment]
lgtm

**Pull request merged**